source/server/listener_manager_impl.cc

```diff
@@ -423,6 +423,33 @@ bool ListenerManagerImpl::addOrUpdateListener(const envoy::config::listener::v3:
   return false;
 }
 
+void ListenerManagerImpl::setupSocketFactoryForListener(ListenerImpl& new_listener,
+                                                        const ListenerImpl& existing_listener) {
+  bool same_socket_options = true;
+  if (Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.enable_update_listener_socket_options")) {
+    if (new_listener.reusePort() != existing_listener.reusePort()) {
+      incListenerCreateFailureStat();
```

Comment:
I added this for the integration test can know the listener update failed https://github.com/envoyproxy/envoy/pull/23189/files#diff-7fe1ad3a088b5d0ea44d9bace0dfa0675d5009bcb538afb778ebad50c802ddeaR1489, But I feel this stat definition isn't very clear for now, it doesn't increase in many listener update failed case.